### ISSUE TITLE
 freqtrade.exchange.common - WARNING - retrying _async_get_candle_history() 

### ISSUE THREAD

**Issue by sidnicholas** [freqtrade/freqtrade#2584]
## Step 1: Have you search for this issue before posting it?

If you have discovered a bug in the bot, please [search our issue tracker](https://github.com/freqtrade/freqtrade/issues?q=is%3Aissue). 
If it hasn't been reported, please create a new issue.

## Step 2: Describe your environment

  * Operating system: Ubuntu 18.04
  * Python Version:python 3.6.8
  * CCXT version: 1.19.25
  * Branch: Develop
  * Last Commit ID: 904a9c5dc7a66ad5e88183aaa64e5ff6b8d6d8d6
  
## Step 3: Describe the problem:

When I run the bot (running in dry run).  I get the following message:
2019-11-28 18:33:44,620 - freqtrade.exchange.common - WARNING - _async_get_candle_history() returned exception: "Could not load ticker history due to RequestTimeout. Message: GET https://openapi-v2.kucoin.com/api/v1/market/candles?symbol=NULS-BTC&type=5min&endAt=1574987613"

I get this message for all the pairs I have in the whitelist

I also get the message when I am running other functions like download-data

Based on another issue submitted, I thought it might be a problem with the exchange.  I changed the exchange and encountered a similar problem.

  ```
freqtrade -c ./config.json --strategy BBRSI
2019-11-28 21:57:29,202 - freqtrade.loggers - INFO - Verbosity set to 0
2019-11-28 21:57:29,203 - freqtrade.configuration.configuration - INFO - Dry run is enabled
2019-11-28 21:57:29,203 - freqtrade.configuration.configuration - INFO - Using DB: "sqlite://"
2019-11-28 21:57:29,203 - freqtrade.configuration.configuration - INFO - Using max_open_trades: 15 ...
2019-11-28 21:57:29,203 - freqtrade.configuration.configuration - INFO - Using user-data directory: /home/sidmnich/freqtrade/user_data ...
2019-11-28 21:57:29,203 - freqtrade.configuration.configuration - INFO - Using data directory: /home/sidmnich/freqtrade/user_data/data/kucoin ...
2019-11-28 21:57:29,204 - freqtrade.configuration.check_exchange - INFO - Checking exchange...
2019-11-28 21:57:29,204 - freqtrade.configuration.check_exchange - WARNING - Exchange "kucoin" is known to the the ccxt library, available for the bot, but not officially supported by the Freqtrade development team. It may work flawlessly (please report back) or have serious issues. Use it at your own discretion.
2019-11-28 21:57:29,204 - freqtrade.configuration.configuration - INFO - Using pairlist from configuration.
2019-11-28 21:57:29,204 - freqtrade.freqtradebot - INFO - Starting freqtrade develop-904a9c5d
2019-11-28 21:57:29,211 - freqtrade.resolvers.iresolver - INFO - Using resolved strategy BBRSI from '/home/sidmnich/freqtrade/user_data/strategies/bbrsi.py'...
2019-11-28 21:57:29,212 - freqtrade.resolvers.strategy_resolver - INFO - Override strategy 'ticker_interval' with value in config file: 5m.
2019-11-28 21:57:29,212 - freqtrade.resolvers.strategy_resolver - INFO - Override strategy 'trailing_stop' with value in config file: False.
2019-11-28 21:57:29,212 - freqtrade.resolvers.strategy_resolver - INFO - Override strategy 'stake_currency' with value in config file: BTC.
2019-11-28 21:57:29,212 - freqtrade.resolvers.strategy_resolver - INFO - Override strategy 'stake_amount' with value in config file: 0.01.
2019-11-28 21:57:29,212 - freqtrade.resolvers.strategy_resolver - INFO - Override strategy 'use_sell_signal' with value in config file: True.
2019-11-28 21:57:29,212 - freqtrade.resolvers.strategy_resolver - INFO - Override strategy 'sell_profit_only' with value in config file: False.
2019-11-28 21:57:29,212 - freqtrade.resolvers.strategy_resolver - INFO - Override strategy 'ignore_roi_if_buy_signal' with value in config file: True.
2019-11-28 21:57:29,212 - freqtrade.resolvers.strategy_resolver - INFO - Strategy using minimal_roi: {'0': 0.14048, '14': 0.06955, '30': 0.02281, '45': 0}
2019-11-28 21:57:29,212 - freqtrade.resolvers.strategy_resolver - INFO - Strategy using ticker_interval: 5m
2019-11-28 21:57:29,212 - freqtrade.resolvers.strategy_resolver - INFO - Strategy using stoploss: -0.2595
2019-11-28 21:57:29,212 - freqtrade.resolvers.strategy_resolver - INFO - Strategy using trailing_stop: False
2019-11-28 21:57:29,212 - freqtrade.resolvers.strategy_resolver - INFO - Strategy using trailing_stop_positive_offset: 0.0
2019-11-28 21:57:29,212 - freqtrade.resolvers.strategy_resolver - INFO - Strategy using trailing_only_offset_is_reached: False
2019-11-28 21:57:29,212 - freqtrade.resolvers.strategy_resolver - INFO - Strategy using process_only_new_candles: False
2019-11-28 21:57:29,212 - freqtrade.resolvers.strategy_resolver - INFO - Strategy using order_types: {'buy': 'limit', 'sell': 'limit', 'stoploss': 'market', 'stoploss_on_exchange': False}
2019-11-28 21:57:29,212 - freqtrade.resolvers.strategy_resolver - INFO - Strategy using order_time_in_force: {'buy': 'gtc', 'sell': 'gtc'}
2019-11-28 21:57:29,212 - freqtrade.resolvers.strategy_resolver - INFO - Strategy using stake_currency: BTC
2019-11-28 21:57:29,212 - freqtrade.resolvers.strategy_resolver - INFO - Strategy using stake_amount: 0.01
2019-11-28 21:57:29,212 - freqtrade.resolvers.strategy_resolver - INFO - Strategy using startup_candle_count: 0
2019-11-28 21:57:29,212 - freqtrade.resolvers.strategy_resolver - INFO - Strategy using use_sell_signal: True
2019-11-28 21:57:29,212 - freqtrade.resolvers.strategy_resolver - INFO - Strategy using sell_profit_only: False
2019-11-28 21:57:29,212 - freqtrade.resolvers.strategy_resolver - INFO - Strategy using ignore_roi_if_buy_signal: True
2019-11-28 21:57:29,212 - freqtrade.resolvers.exchange_resolver - INFO - No Kucoin specific subclass found. Using the generic class instead.
2019-11-28 21:57:29,212 - freqtrade.exchange.exchange - INFO - Instance is running with dry_run enabled
2019-11-28 21:57:29,212 - freqtrade.exchange.exchange - INFO - Applying additional ccxt config: {'enableRateLimit': True}
2019-11-28 21:57:29,214 - freqtrade.exchange.exchange - INFO - Applying additional ccxt config: {'enableRateLimit': True, 'rateLimit': 500}
2019-11-28 21:57:29,216 - freqtrade.exchange.exchange - INFO - Using Exchange "KuCoin"
2019-11-28 21:57:36,775 - freqtrade.wallets - INFO - Wallets synced.
2019-11-28 21:57:36,775 - freqtrade.resolvers.iresolver - WARNING - Path "/home/sidmnich/freqtrade/user_data/pairlist" does not exist.
2019-11-28 21:57:36,776 - freqtrade.resolvers.iresolver - INFO - Using resolved pairlist StaticPairList from '/home/sidmnich/freqtrade/freqtrade/pairlist/StaticPairList.py'...
2019-11-28 21:57:36,783 - freqtrade.rpc.rpc_manager - INFO - Enabling rpc.telegram ...
2019-11-28 21:57:37,624 - freqtrade.rpc.telegram - INFO - rpc.telegram is listening for following commands: [['status'], ['profit'], ['balance'], ['start'], ['stop'], ['forcesell'], ['forcebuy'], ['performance'], ['daily'], ['count'], ['reload_conf'], ['stopbuy'], ['whitelist'], ['blacklist'], ['edge'], ['help'], ['version']]
2019-11-28 21:57:38,290 - freqtrade.rpc.rpc_manager - INFO - Sending rpc message: {'type': status, 'status': 'running'}
2019-11-28 21:57:38,649 - freqtrade.worker - INFO - Changing state to: RUNNING
2019-11-28 21:57:38,649 - freqtrade.rpc.rpc_manager - INFO - Sending rpc message: {'type': warning, 'status': 'Dry run is enabled. All trades are simulated.'}
2019-11-28 21:57:38,937 - freqtrade.rpc.rpc_manager - INFO - Sending rpc message: {'type': custom, 'status': "*Exchange:* `kucoin`\n*Stake per trade:* `0.01 BTC`\n*Minimum ROI:* `{'0': 0.14048, '14': 0.06955, '30': 0.02281, '45': 0}`\n*Stoploss:* `-0.2595`\n*Ticker Interval:* `5m`\n*Strategy:* `BBRSI`"}
2019-11-28 21:57:39,650 - freqtrade.rpc.rpc_manager - INFO - Sending rpc message: {'type': status, 'status': "Searching for BTC pairs to buy and sell based on StaticPairList: ['ADA/BTC', 'ETC/BTC', 'ETH/BTC', 'LSK/BTC', 'ZRX/BTC', 'WAXP/BTC', 'DCR/BTC', 'ENJ/BTC', 'TRX/BTC', 'XMR/BTC', 'DGB/BTC', 'MANA/BTC', 'BCH/BTC', 'XEM/BTC', 'PIVX/BTC', 'LOOM/BTC', 'XRP/BTC', 'NEO/BTC', 'DASH/BTC', 'XLM/BTC', 'ZEC/BTC', 'SPND/BTC', 'LTC/BTC']"}
2019-11-28 21:57:50,619 - freqtrade.exchange.common - WARNING - _async_get_candle_history() returned exception: "Could not load ticker history due to RequestTimeout. Message: GET https://openapi-v2.kucoin.com/api/v1/market/candles?symbol=ADA-BTC&type=5min&endAt=1574999860"
2019-11-28 21:57:50,620 - freqtrade.exchange.common - WARNING - retrying _async_get_candle_history() still for 3 times
2019-11-28 21:57:50,620 - freqtrade.exchange.common - WARNING - _async_get_candle_history() returned exception: "Could not load ticker history due to RequestTimeout. Message: GET https://openapi-v2.kucoin.com/api/v1/market/candles?symbol=ENJ-BTC&type=5min&endAt=1574999860"
2019-11-28 21:57:50,620 - freqtrade.exchange.common - WARNING - retrying _async_get_candle_history() still for 3 times
2019-11-28 21:57:50,620 - freqtrade.exchange.common - WARNING - _async_get_candle_history() returned exception: "Could not load ticker history due to RequestTimeout. Message: GET https://openapi-v2.kucoin.com/api/v1/market/candles?symbol=TRX-BTC&type=5min&endAt=1574999860"
2019-11-28 21:57:50,621 - freqtrade.exchange.common - WARNING - retrying _async_get_candle_history() still for 3 times
2019-11-28 21:57:51,620 - freqtrade.exchange.common - WARNING - _async_get_candle_history() returned exception: "Could not load ticker history due to RequestTimeout. Message: GET https://openapi-v2.kucoin.com/api/v1/market/candles?symbol=DGB-BTC&type=5min&endAt=1574999860"
2019-11-28 21:57:51,620 - freqtrade.exchange.common - WARNING - retrying _async_get_candle_history() still for 3 times
2019-11-28 21:57:51,620 - freqtrade.exchange.common - WARNING - _async_get_candle_history() returned exception: "Could not load ticker history due to RequestTimeout. Message: GET https://openapi-v2.kucoin.com/api/v1/market/candles?symbol=SPND-BTC&type=5min&endAt=1574999860"
2019-11-28 21:57:51,620 - freqtrade.exchange.common - WARNING - retrying _async_get_candle_history() still for 3 times
2019-11-28 21:57:52,619 - freqtrade.exchange.common - WARNING - _async_get_candle_history() returned exception: "Could not load ticker history due to RequestTimeout. Message: GET https://openapi-v2.kucoin.com/api/v1/market/candles?symbol=ETC-BTC&type=5min&endAt=1574999860"
2019-11-28 21:57:52,620 - freqtrade.exchange.common - WARNING - retrying _async_get_candle_history() still for 3 times
2019-11-28 21:57:52,620 - freqtrade.exchange.common - WARNING - _async_get_candle_history() returned exception: "Could not load ticker history due to RequestTimeout. Message: GET https://openapi-v2.kucoin.com/api/v1/market/candles?symbol=XLM-BTC&type=5min&endAt=1574999860"
2019-11-28 21:57:52,620 - freqtrade.exchange.common - WARNING - retrying _async_get_candle_history() still for 3 times
2019-11-28 21:57:53,619 - freqtrade.exchange.common - WARNING - _async_get_candle_history() returned exception: "Could not load ticker history due to RequestTimeout. Message: GET https://openapi-v2.kucoin.com/api/v1/market/candles?symbol=DCR-BTC&type=5min&endAt=1574999860"
2019-11-28 21:57:53,619 - freqtrade.exchange.common - WARNING - retrying _async_get_candle_history() still for 3 times
2019-11-28 21:57:53,620 - freqtrade.exchange.common - WARNING - _async_get_candle_history() returned exception: "Could not load ticker history due to RequestTimeout. Message: GET https://openapi-v2.kucoin.com/api/v1/market/candles?symbol=LSK-BTC&type=5min&endAt=1574999860"
2019-11-28 21:57:53,620 - freqtrade.exchange.common - WARNING - retrying _async_get_candle_history() still for 3 times
^C2019-11-28 21:57:54,039 - freqtrade - INFO - SIGINT received, aborting ...
2019-11-28 21:57:54,039 - freqtrade.rpc.rpc_manager - INFO - Sending rpc message: {'type': status, 'status': 'process died'}
2019-11-28 21:57:57,273 - freqtrade.freqtradebot - INFO - Cleaning up modules ...
2019-11-28 21:57:57,273 - freqtrade.rpc.rpc_manager - INFO - Cleaning up rpc modules ...
  ```

**Comment by xmatthias** [comment]
I've just tried this - and it does work flawlessly for me using kucoin.

there's a few things which could cause this:
* your bot is located in a country kucoin does not support (i mention bot instead of yourself, assuming it could run on a server in another country)
* the exchange was temporarily down when you tried, but works now (when i tried) - so please retry
* something is wrong with networking - best try to go to one of the mentioned url's above [for example](https://openapi-v2.kucoin.com/api/v1/market/candles?symbol=XLM-BTC&type=5min&endAt=1574999860) and check if it returns errors
* you have a proxy configured (either system-wide, or within the bot) which does not properly work (or which is blocked by the exchange)

Since you mention it also did not work for another exchange, i would assume it's either a country blocking thing - or a proxy / networking thing - which may be difficult to debug for us.

Hope this gives you some ideas where to search for - however i don't see this problem in freqtrade (since it works for me ...).

**Comment by sidnicholas** [comment]
Thanks,

I am still having problems with Kucoin although I was able to run intermittently on Binance US.  I think there may be some network problems on my end that I will need to resolve.

Thanks for the support,

We can go ahead and close this ticket out.
